### DISCUSSION TITLE
Sphinx Pythia Theme Updated! Check this out for important information

### DISCUSSION THREAD

**Discussion by erogluorhan** [ProjectPythia/projectpythia.github.io#420] [General]
Dear Pythia community! 

Now as we have updated the outdated [sphinx-pythia-theme](https://github.com/ProjectPythia/sphinx-pythia-theme) and released it on [Conda](https://anaconda.org/conda-forge/sphinx-pythia-theme) and [PyPI](https://pypi.org/project/sphinx-pythia-theme/), we'd like to share some important information about it and ask your feedback below:

1. [We ask your feedback](https://github.com/ProjectPythia/projectpythia.github.io/discussions/420#discussioncomment-8960607)
2. [How to transition existing cookbooks](https://github.com/ProjectPythia/projectpythia.github.io/discussions/420#discussioncomment-8960608)
3. [How to write new cookbooks](https://github.com/ProjectPythia/projectpythia.github.io/discussions/420#discussioncomment-8960626)

**Comment by erogluorhan** [comment]
## We ask your feedback:

First of all, this theme update is a huge jump from Sphinx<5 to Sphinx >=7 that covers a lot of significant infrastructural change, which we tried to limit its impact on Pythia's user-facing online assets (portal, cookbooks, foundations, etc.). Though, as with any huge theme update like this, there might be functional and cosmetic etc. issues. 

Please add a comment below in this cell if you observe any issues:

**Reply by erogluorhan** [reply]
Let me go first:

[Dropdown buttons in the Resource Gallery and Cookbook Gallery are not working](https://github.com/ProjectPythia/projectpythia.github.io/issues/419).

**Reply by erogluorhan** [reply]
[In mobile version of the portal, navigation bar items are not showing up](https://github.com/ProjectPythia/projectpythia.github.io/issues/430)

**Reply by r-ford** [reply]
Code blocks are difficult to read in dark mode: 
![image](https://github.com/ProjectPythia/projectpythia.github.io/assets/50349951/c2f4bf3c-6b37-4247-9d1f-95faf2af935d)

**Comment by erogluorhan** [comment]
f## How to transition existing cookbooks:

We have already transitioned the [cookbook-template](https://github.com/ProjectPythia/cookbook-template) into the updated theme in this pull request: [Start using the updated theme](https://github.com/ProjectPythia/cookbook-template/pull/163).

Here is a checklist for this transition of the existing cookbooks:

- Write something next to the cookbook's name below to show you will be / are working on it (GitHub username, issue or PR link, etc.)
- Mark it when completed
- Enter a comment below if you have any issues etc.

Checklist:

- [x] [CESM LENS on AWS Cookbook](https://projectpythia.org/cesm-lens-aws-cookbook/README.html) (Still issues after this [PR](https://github.com/ProjectPythia/cesm-lens-aws-cookbook/pull/23), see below comment)) ([new PR](https://github.com/ProjectPythia/cesm-lens-aws-cookbook/pull/24))
- [x] [CMIP6 Cookbook](https://projectpythia.org/cmip6-cookbook/README.html) (@jukent )
- [x] [HRRR AWS Cookbook](https://projectpythia.org/HRRR-AWS-cookbook/README.html) (@jukent)
- [x] [Radar Cookbook](https://projectpythia.org/radar-cookbook/README.html) (@jukent) ([broken](https://github.com/ProjectPythia/radar-cookbook/actions/runs/8526722286/job/23356581129))
- [x] [Intake Cookbook](https://projectpythia.org/intake-cookbook/README.html) (@jukent) ([PR](https://github.com/ProjectPythia/intake-cookbook/pull/34))
- [x] [Landsat ML Cookbook](https://projectpythia.org/landsat-ml-cookbook/README.html) (@jukent)
- [x] [Kerchunk Cookbook](https://projectpythia.org/kerchunk-cookbook/README.html) (@jukent)
- [x] [xbatcher for Machine Learning Part 1 Cookbook](https://projectpythia.org/xbatcher-ML-1-cookbook/README.html) (@jukent) ([PR](https://github.com/ProjectPythia/xbatcher-ML-1-cookbook/pull/14))
- [x] [Dask Cookbook](https://projectpythia.org/dask-cookbook/README.html) ([PR](https://github.com/ProjectPythia/dask-cookbook/pull/17))
- [x] [ARCO ERA-5 Interactive Visualization Cookbook](https://projectpythia.org/ERA5_interactive-cookbook/README.html) ([PR](https://github.com/ProjectPythia/ERA5_interactive-cookbook/pull/10))
- [x] [Web Map / Feature Services Cookbook](https://projectpythia.org/web-map-feature-services-cookbook/README.html) (@jukent) ([PR](https://github.com/ProjectPythia/web-map-feature-services-cookbook/pull/9)) failing link check
- [x] [Sentinel-2 L2A Interactive Dashboard](https://projectpythia.org/interactive-sentinel-2-cookbook/README.html) (@jukent) ([PR](https://github.com/ProjectPythia/interactive-sentinel-2-cookbook/pull/8))
- [x] [(re)Gridding with xarray](https://projectpythia.org/gridding-cookbook/README.html) (@jukent) ([PR](https://github.com/ProjectPythia/gridding-cookbook/pull/21))
- [x] [Advanced Visualization Cookbook](https://projectpythia.org/advanced-viz-cookbook/README.html) (@jukent) ([PR](https://github.com/ProjectPythia/advanced-viz-cookbook/pull/53))
- [x] [Unstructured Grids Visualization Cookbook](https://projectpythia.org/unstructured-grid-viz-cookbook/README.html) ([PR](https://github.com/ProjectPythia/unstructured-grid-viz-cookbook/pull/10))
- [x] [NA-CORDEX Visualization Cookbook](https://projectpythia.org/na-cordex-viz-cookbook/README.html) (@jukent) ([PR](https://github.com/ProjectPythia/na-cordex-viz-cookbook/pull/14))

**Reply by erogluorhan** [reply]
[CESM LENS on AWS Cookbook](https://projectpythia.org/cesm-lens-aws-cookbook/README.html) - Merged this [PR](https://github.com/ProjectPythia/cesm-lens-aws-cookbook/pull/23) but CI [didn't pass](https://github.com/ProjectPythia/cesm-lens-aws-cookbook/actions/runs/8489515014). This cookbook is supposed to be generated via BInder, and there seems to be an issue with it when the workflow is dispatched manually or after a PR merge. Maybe tonight's auto nightly-build will pass.

**Reply by jukent** [reply]
[Intake Cookbook](https://github.com/ProjectPythia/intake-cookbook/pull/34) and [Xbatcher](https://github.com/ProjectPythia/xbatcher-ML-1-cookbook/pull/14) are failing for reasons other than the theme that I was not able to fix today.

**Reply by jukent** [reply]
[CESM LENS fixed](https://github.com/ProjectPythia/cesm-lens-aws-cookbook/actions/runs/8517866237/job/23329081616)!

**Reply by jukent** [reply]
Radar cookbook theme is merged, [PR to address failing build](https://github.com/ProjectPythia/radar-cookbook/pull/123). Merged because it passed the PR tests, but is failing locally and [to publish](https://github.com/ProjectPythia/radar-cookbook/actions/runs/8526722286/job/23356581129). @mgrover1 could you or someone more familiar with the radar packages look at this?

**Reply by mgrover1** [reply]
I fixed the radar cookbook with the latest env - an older version of a package was cached.

**Reply by jukent** [reply]
Should we merge the PRs for the new theme even if the build is failing for other reasons? Only Intake, Xbatcher, and EOFs are still failing.

@cmdupuis3 @mgrover1 @mgrover1 or @r-ford do any of you feel confident addressing the failure in Xbatcher?

**Reply by cmdupuis3** [reply]
@jukent So atm, the xbatcher cookbook doesn't actually run xbatcher by default, so that shouldn't be an issue. I believe the hangup is mostly around data access, although that code was relying on intake as well. In any case, I think the plan was to wait for the data allocation to open up so I can put the needed files on it.

**Reply by jukent** [reply]
Great thanks. I'll merge only the theme changes in that case.

**Comment by erogluorhan** [comment]
## How to write new cookbooks:

As before, use our [cookbook-template](https://github.com/ProjectPythia/cookbook-template) that we have already revised to use the updated new theme. Please see the [Project Pythia Cookbook Contributor’s Guide](https://github.com/ProjectPythia/.github/blob/main/.github/CONTRIBUTING.md) for further information.

**Comment by jukent** [comment]
More Cookbooks that need updating (some on gallery, some not):

- [x] [VAPOR](https://github.com/ProjectPythia/vapor-python-cookbook)
- [x] [EOFs](https://github.com/ProjectPythia/eofs-cookbook) ([PR](https://github.com/ProjectPythia/eofs-cookbook/pull/10)) broken
- [x] [Marine Heatwave](https://github.com/ProjectPythia/marine-heatwave-cookbook)
- [x] [MetPy](https://github.com/ProjectPythia/metpy-cookbook)
- [x] [XGCM](https://github.com/ProjectPythia/xgcm-cookbook)
- [x] [Physical Oceanography](https://github.com/ProjectPythia/physical-oceanography-cookbook)
- [x] [Packaging](https://github.com/ProjectPythia/packaging-cookbook)

**Reply by jukent** [reply]
@r-ford could you look more at why the EOFs cookbook is failing?

**Reply by r-ford** [reply]
Probably related to recent changes in the xeofs package, so I'll try to fix it soon.

**Comment by brian-rose** [comment]
Our [Cook-off site](https://projectpythia.org/pythia-cookoff-2024/) also needs updating to the new theme. It is currently [failing its nightly build](https://github.com/ProjectPythia/pythia-cookoff-2024/actions/runs/8655351658/job/23734160246). I will address this shortly.

Update: [here's the PR](https://github.com/ProjectPythia/pythia-cookoff-2024/pull/5)
